src/mpi-base.jl

```diff
@@ -156,6 +120,54 @@ function Comm_size(comm::Comm)
     Int(size[])
 end
 
+function Type_create_struct{T <: Any}(::Type{T})  # <: Any effectively 
```

Comment:
This function will be called at most once for each type `T`, so the compiler does not need to optimize it -- that would take more time than can be saved.